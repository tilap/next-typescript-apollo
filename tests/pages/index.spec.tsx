import { screen } from '@testing-library/react';
import { useRouter } from 'next/router';
import Index from 'pages/index';
import { renderWithApollo } from '../renderers';

jest.mock('next/router', () => ({
  useRouter: jest.fn(),
}));

jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: (key: string) => key }),
}));

describe('Page Index', () => {
  describe('Page does not change', () => {
    it('match previous snapshot', () => {
      (useRouter as jest.Mock).mockImplementation(() => ({
        locale: 'en',
        locales: ['en'],
      }));
      const result = renderWithApollo(<Index />);
      expect(result.baseElement).toMatchSnapshot();
    });
  });

  describe('Page shows button to switch language', () => {
    it('does not render button when there is only one language available', () => {
      (useRouter as jest.Mock).mockImplementation(() => ({
        locale: 'en',
        locales: ['en'],
      }));
      renderWithApollo(<Index />);

      expect(screen.queryByTestId('changelanguage-btn-en')).not.toBeInTheDocument();
      expect(screen.queryByTestId('changelanguage-btn-ru')).not.toBeInTheDocument();
    });

    it('renders one button when another language is available', () => {
      (useRouter as jest.Mock).mockImplementation(() => ({
        locale: 'en',
        locales: ['en', 'ru'],
      }));
      renderWithApollo(<Index />);

      expect(screen.queryByTestId('changelanguage-btn-en')).not.toBeInTheDocument();
      expect(screen.queryByTestId('changelanguage-btn-ru')).toBeInTheDocument();
    });

    it('renders many buttons when another langages are available', () => {
      const push = jest.fn();
      (useRouter as jest.Mock).mockImplementation(() => ({
        push,
        pathname: '/',
        route: '/',
        asPath: '/',
        query: '',
        locale: 'fr',
        locales: ['en', 'ru', 'fr', 'nl'],
      }));
      renderWithApollo(<Index />);

      expect(screen.queryByTestId('changelanguage-btn-fr')).not.toBeInTheDocument();
      expect(screen.queryByTestId('changelanguage-btn-en')).toBeInTheDocument();
      expect(screen.queryByTestId('changelanguage-btn-ru')).toBeInTheDocument();
      expect(screen.queryByTestId('changelanguage-btn-nl')).toBeInTheDocument();
    });
  });
});
