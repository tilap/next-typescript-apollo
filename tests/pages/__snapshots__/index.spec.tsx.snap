// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page Index Page does not change match previous snapshot 1`] = `
<body>
  <div>
    <div>
      <p>
        greetings
      </p>
    </div>
  </div>
</body>
`;
